### PR TITLE
WIP - Tekton example read me

### PR DESCRIPTION
**Pull request by chilanti**
Initial take - to be completed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,73 @@
...
+
+## Introduction
+This repo contains an example of a Tekton pipeline that builds and deploys an application created with [Appsody](https://github.com/appsody/appsody) to a Kubernetes cluster. The application is deployed as a Knative Serving service. 
+
+## Prerequisites and assumptions
+This example can be put to fruition once you have fulfilled the following prerequisites:
+1) You have access to a Kubernetes cluster where Knative and its prerequisites are configured. You can read about setting up Knative [here](https://knative.dev/docs/install/).
+2) Your Kubernetes cluster has the [Tekton pipelines installed](https://github.com/tektoncd/pipeline/blob/master/docs/install.md).
+3) You have created an application using the appsody CLI, and have checked in your code in a GitHub repository.
+4) Your code repository includes a Knative Serving manifest file called `appsody-service.yaml`. We'll discuss this aspect more in detail later on.
```
Comment (helenmasters) [inline README.md]: "We'll discuss this aspect in more detail later on."

File: README.md [hunk 2]
```diff
@@ -1,2 +1,73 @@
...
+Example implementation of a Tekton pipeline that deploys an Appsody project.
+
+## Introduction
+This repo contains an example of a Tekton pipeline that builds and deploys an application created with [Appsody](https://github.com/appsody/appsody) to a Kubernetes cluster. The application is deployed as a Knative Serving service. 
+
+## Prerequisites and assumptions
+This example can be put to fruition once you have fulfilled the following prerequisites:
+1) You have access to a Kubernetes cluster where Knative and its prerequisites are configured. You can read about setting up Knative [here](https://knative.dev/docs/install/).
+2) Your Kubernetes cluster has the [Tekton pipelines installed](https://github.com/tektoncd/pipeline/blob/master/docs/install.md).
+3) You have created an application using the appsody CLI, and have checked in your code in a GitHub repository.
```
Comment (helenmasters) [inline README.md]: "...using the appsody CLI, and your code is in a GitHub repository."

File: README.md [hunk 3]
```diff
@@ -1,2 +1,73 @@
 # tekton-example
-Example implementation of a tekton pipeline that deploys an Appsody project.
+Example implementation of a Tekton pipeline that deploys an Appsody project.
+
+## Introduction
+This repo contains an example of a Tekton pipeline that builds and deploys an application created with [Appsody](https://github.com/appsody/appsody) to a Kubernetes cluster. The application is deployed as a Knative Serving service. 
+
+## Prerequisites and assumptions
+This example can be put to fruition once you have fulfilled the following prerequisites:
```
Comment (helenmasters) [inline README.md]: I'd like to lose the wording "This example can be put to fruition" perhaps... "In order to run this example, the following prerequisites are required:" or similar?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,73 @@
...
+
+1) Since the Tekton pipeline needs to deploy to the cluster itself, you want to ensure that it runs under an identity that has cluster administrator privileges. 
+
+For this reason, create a service account and the appropriate cluster role binding by issuing:
+```
+kubectl apply -f appsody-service-account.yaml
+kubectl apply -f appsody-cluster-role-binding.yaml
+```
+
+2) Now, create the pipeline task and the pipeline definition. We have here a simple pipeline, with just a single task that performs the various steps of building and deploying the project:
```
Comment (helenmasters) [inline README.md]: could be simplified to "We have a simple pipeline, with a single task..."

File: README.md [hunk 5]
```diff
@@ -1,2 +1,73 @@
...
+```
+kubectl apply -f appsody-build-task.yaml
+kubectl apply -f appsody-build-pipeline.yaml
+```
+3) The pipeline requires the definition of two resources in order to operate:
+* The definition docker image that is built and deployed by the pipeline itself
```
Comment (helenmasters) [inline README.md]: docker should be Docker?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,73 @@
...
+    type: git
+```
+4) Once you have edited the resources, apply them to your cluster:
+```
+kubectl apply -f appsody-pipeline-resources.yaml
+```
+The Tekton pipeline is now fully set up.
+
+## A few words on the required deployment manifest
+<TBD>
```
Comment (helenmasters) [inline README.md]: README looks good once this <TBD> section has been completed.

File: README.md (hunk 3 above)
Comment (chilanti) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (chilanti) [reply README.md]: Done.

File: README.md (hunk 6 above)
Comment (chilanti) [reply README.md]: Completed the section - please take a look. Thanks.

**Comment by chilanti** [comment]
@helenmasters - I did complete the missing section - ready to review and merge.

**Pull request merged**